dm-dedup-rw.c

```diff
@@ -64,15 +66,19 @@ static int merge_data(struct dedup_config *dc, struct page *page,
 		goto out;
 	}
 
-	src_page_vaddr = page_address(bio->bi_io_vec->bv_page);
-	dest_page_vaddr = page_address(page);
-
-	src_page_vaddr = src_page_vaddr + bio->bi_io_vec->bv_offset;
 	/* Locating the right sector to merge */
-	dest_page_vaddr = dest_page_vaddr + (to_bytes(position));
+	dest_page_vaddr = page_address(page) + to_bytes(position);
+
+	bio_for_each_segment(bvec, bio, iter) {
+		src_page_vaddr = bio_data(bio);
+
+		/* Merging Data */
+		memmove(dest_page_vaddr, src_page_vaddr, bio_cur_bytes(bio));
+
+		/* Updating destination address */
+		dest_page_vaddr += bio_cur_bytes(bio);
+    }
```

Comment:
So if I understand correctly, you are saying that I should get the page and cur_bytes from this copy iter instead of the bio's iter. Right?